00_Base/src/repository/OcpiLocationRepository.ts

```diff
@@ -77,6 +77,54 @@ export class OcpiLocationRepository extends SequelizeRepository<OcpiLocation> {
     );
   }
 
+  async createOrUpdateOcpiLocation(
+    location: Partial<OcpiLocation>,
+  ): Promise<OcpiLocation> {
+    if (location[OcpiLocationProps.citrineLocationId]) {
+      const [savedOcpiLocation, ocpiLocationCreated] =
+        await this._readOrCreateByQuery({
+          where: {
+            [OcpiLocationProps.citrineLocationId]:
+              location[OcpiLocationProps.citrineLocationId],
+          },
+          defaults: {
```

Comment:
yup exactly -- i'm using readOrCreateByQuery to have some shortcuts since we only want to create if it doesn't exist, otherwise just simply update the object.